### PR TITLE
fix: skip nil strategies in wait.ForAll

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR skips nil wait strategies in case some downstream consumer uses the wait.ForAll without checking the validity of all the straegies in the list.

A clear example is building a list of strategies depending on certain conditions, and using forAll multiple times.

## Why is it important?
Avoid runtime panics

### CODE REVIEW

File: wait/all.go [hunk 1]
```diff
@@ -62,6 +62,13 @@ func (ms *MultiStrategy) WaitUntilReady(ctx context.Context, target StrategyTarg
 	}
 
 	for _, strategy := range ms.Strategies {
+		if strategy == nil {
```
Comment (stevenh) [inline wait/all.go:65]: bug: this won't detect a nil type which implements Strategy see https://go.dev/play/p/yz46T3tUXOR

File: wait/all_test.go [hunk 2]
```diff
@@ -124,3 +124,27 @@ func TestMultiStrategy_WaitUntilReady(t *testing.T) {
 		})
 	}
 }
+
+func TestMultiStrategy_handleNils(t *testing.T) {
```
Comment (stevenh) [inline wait/all_test.go:128]: bug: missing test for nil type which implements Strategy.

**Review by stevenh — CHANGES_REQUESTED** [review]

**Review by stevenh — CHANGES_REQUESTED** [review]
Looking good just the lint failures to fix.

**Review by stevenh — APPROVED** [review]
